kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/Line.kt

```diff
@@ -1,6 +1,15 @@
 package scientifik.kmath.geometry
 
-data class Line<V: Vector>(val base: V, val direction: V)
+import scientifik.kmath.dimensions.D2
+import scientifik.kmath.dimensions.D3
+import scientifik.kmath.dimensions.Dimension
 
-typealias Line2D = Line<Vector2D>
-typealias Line3D = Line<Vector3D>
+// TODO think how we can add asserts about direction.norm() > 0
+data class Line<T, D : Dimension, V : Vector<T, D>>(val base: V, val direction: V)
+
+// TODO assert that p1 != p2
+fun <T, D : Dimension, V : Vector<T, D>> VectorSpace<T, D, V>.lineThrough(p1: V, p2: V): Line<T, D, V> =
+        Line(p1, with(this) { p2 - p1 })
```

Comment:
with(this) makes no sense, replace with `Line(p1, p2 - p1)`

Reply:
agree, removed